GraphAlgorithmsTests/MaxCliqueNaiveFinderTests.cs

```diff
@@ -0,0 +1,207 @@
+            //
+
+            // 1) v = 3: 3 = deg(v) == MAX_CLIQUE = 3
+            graph = new Graph(5);
+            graph.AdjacencyMatrix = new int[,] {
+                { 0, 1, 1, 1, 0 },
+                { 1, 0, 1, 1, 0 },
+                { 1, 1, 0, 1, 0 },
+                { 1, 1, 0, 0, 1 },
+                { 0, 0, 0, 1, 0 },
+            };
+            yield return new TestCaseData(graph, new List<int> { 0, 1, 2 });
+
+            // 2) v = 3 : 4 = deg(v) > MAX_CLIQUE = 3
+            graph = new Graph(6);
+            graph.AdjacencyMatrix = new int[,] {
+                { 0, 1, 1, 1, 0, 0 },
+                { 1, 0, 1, 0, 0, 0 },
+                { 1, 1, 0, 1, 0, 0 },
+                { 1, 0, 1, 0, 1, 1 },
+                { 0, 0, 0, 1, 0, 0 },
+                { 0, 0, 0, 1, 0, 0 },
+            };
+            yield return new TestCaseData(graph, new List<int> { 0, 1, 2 });
+
+            // some random graphs, graphs in which vertices 'v' that don't belong to clique have
+            // deg(v) < MAX_CLIQUE
+            var random = new Random();
+            int maxSize = 25;
+            for (int i = 0; i < 100; ++i)
+            {
+                graph = new Graph(random.Next(1, maxSize));
+                int cliqueSize = random.Next(1, graph.Size);
+                var clique = GenerateClique(graph, cliqueSize);
+                GenerateAdditionalEdges(graph, clique);
+                yield return new TestCaseData(graph, clique);
+            }
+        }
+
+        private static List<int> GenerateClique(Graph graph, int cliqueSize)
+        {
+            if (cliqueSize == 1)
+                return new List<int>() { 0 };
+            Random random = new Random();
+            List<int> clique = new List<int>();
+            List<int> availableVertices = Enumerable.Range(0, graph.Size).ToList();
+            while (clique.Count < cliqueSize)
+            {
+                int index = random.Next(availableVertices.Count);
+                int u = availableVertices[index];
+                availableVertices.RemoveAt(index);
+                foreach (var v in clique)
+                    graph.AddBothSidesEdge(u, v);
+                clique.Add(u);
+            }
+            return clique;
+        }
+
+        private static void GenerateAdditionalEdges(Graph graph, List<int> clique)
+        {
+            Random random = new Random();
+            int add = random.Next(graph.Size);
+            for (int i = 0; i < add; ++i)
+            {
+                int u = random.Next(graph.Size);
+                int v = random.Next(graph.Size);
+                if (graph.GetDegree(u) < clique.Count - 1 && graph.GetDegree(v) < clique.Count - 1)
+                    graph.AddEdge(u, v);
+            }
+        }
+
+        [Test]
+        [TestCaseSource(nameof(TestCases))]
+        public void FindTest(Graph graph, List<int> expectedResult)
+        {
+            List<int> result = MaxCliqueNaiveFinder.Find(graph);
```

Comment:
Maybe just check the number of edges. Or write a method that checks if the returned graph is a clique

Reply:
I'm leaving that up to @DybowskiB to decide